python/semantic_kernel/semantic_functions/chat_with_data_prompt_template.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from logging import Logger
+from typing import TYPE_CHECKING, List, Optional, TypeVar
+
+from semantic_kernel.models.chat.chat_message import ChatMessage
+from semantic_kernel.semantic_functions.chat_prompt_template import ChatPromptTemplate
+from semantic_kernel.semantic_functions.prompt_template_config import (
+    PromptTemplateConfig,
+)
+from semantic_kernel.template_engine.protocols.prompt_templating_engine import (
+    PromptTemplatingEngine,
+)
+
+if TYPE_CHECKING:
+    from semantic_kernel.orchestration.sk_context import SKContext
+
+ChatMessageT = TypeVar("ChatMessageT", bound=ChatMessage)
+
+
+class ChatWithDataPromptTemplate(ChatPromptTemplate):
+    _messages: List[ChatMessageT]
+
+    def __init__(
```

Comment:
nit: could we add a doctstring here, please? It's helpful to provide context to readers/users.